lib/ydeploy.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+class rex_ydeploy
+{
+    private static $instance;
+
+    private $deployed;
+
+    private $host;
+    private $stage;
+    private $commit;
+    private $timestamp;
+
+    private function __construct()
+    {
+        $path = rex_path::addonData('ydeploy', 'info.json');
+
+        if (!file_exists($path)) {
+            $this->deployed = false;
+
+            return;
+        }
+
+        $this->deployed = true;
+
+        $info = rex_file::getCache($path);
+
+        $this->host = $info['host'];
+        $this->stage = $info['stage'];
+        $this->commit = $info['commit'];
+        $this->timestamp = DateTimeImmutable::createFromFormat('U', $info['timestamp']);
+    }
+
+    public static function factory(): self
+    {
+        if (self::$instance) {
+            return self::$instance;
+        }
+
+        return self::$instance = new self();
+    }
+
+    /**
+     * Returns whether this a deployed or local instance.
+     *
+     * @return bool
+     */
+    public function isDeployed(): bool
+    {
+        return $this->deployed;
+    }
+
+    /**
+     * Returns the host name (from deploy.php).
```

Comment:
warum „from deploy.php“?

Reply:
Weil es nicht etwa `$_SERVER['host']` oder irgend sowas ist, sondern der Host, so wie er in der deploy.php benannt wurde.